core/feconf.py

```diff
@@ -24,11 +24,31 @@
 
 from core.constants import constants
 
-from typing import Dict, List, Union
+from typing import Dict, List
 from typing_extensions import TypedDict
 
-CommandType = (
-    Dict[str, Union[str, List[str], Dict[str, Union[str, List[str]]]]])
+
+# This TypedDict is used intentionally so that the type checker can have
+# any of the following keys ommited.
+class ValidCmdAttributeSpecsAttributeDictNotTotal(TypedDict, total=False):
```

Comment:
```suggestion
# This TypedDict is used only to create the TypedDict below, because we want 
# to mark some attributes as optional.
class _ValidCmdAttributeSpecsAttributeDictNotTotal(TypedDict, total=False):
```

Not sure if the `_` will be accepted in from of the class name.

Reply:
Done!